cmake/extensions.cmake

```diff
@@ -126,8 +126,29 @@ endfunction()
 # merges the resulting hex files. The 'flash' target is updated to use this
 # merged hex file instead of the 'zephyr.hex' file from the origin context.
 #
-function(zephyr_add_executable name)
-  set_property(GLOBAL PROPERTY IMAGE ${name}_)
+function(zephyr_add_executable name require_build)
+  set(${require_build} 0)
+  string(TOUPPER ${name} UPNAME)
+
+  if (${CONFIG_${UPNAME}_USE_HEX_FILE})
+    if (NOT EXISTS ${CONFIG_${UPNAME}_HEX_FILE})
```

Comment:
```
assert_exists(CONFIG_${UPNAME}_HEX_FILE)
message("Using ${CONFIG_${UPNAME}_HEX_FILE} instead of building ${name}")
set_property(GLOBAL APPEND PROPERTY
[...]
```